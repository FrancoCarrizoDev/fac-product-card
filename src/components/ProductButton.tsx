import { useCallback, useContext } from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css';
import React from 'react';

export interface Props {
  className?: string;
  style?: React.CSSProperties;
}

export const ProductButtons = ({ className, style }: Props) => {
  const { increasyBy, counter, maxCount } = useContext(ProductContext);

  const isMaxReached = useCallback(() => !!maxCount && counter === maxCount, [
    counter,
    maxCount,
  ]);

  return (
    <div className={`${styles.buttonsContainer} ${className}`} style={style}>
      <button className={styles.buttonMinus} onClick={() => increasyBy(-1)}>
        {' '}
        -{' '}
      </button>

      <div className={styles.countLabel}>{counter}</div>

      <button
        className={`${isMaxReached() ? styles.disabled : ''} ${
          styles.buttonAdd
        }`}
        onClick={() => increasyBy(+1)}
      >
        {' '}
        +{' '}
      </button>
    </div>
  );
};
