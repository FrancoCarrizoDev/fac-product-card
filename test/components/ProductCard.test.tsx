import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';

const { act } = renderer;

describe('ProductTitle', () => {
  test('Debe de mostrar el componente correctament', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <h1>ProductCard</h1>}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe de incrementar el contador', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({ count, increasyBy }) => (
          <>
            <h1>ProductCard</h1>
            <span>{count}</span>
            <button onClick={() => increasyBy(1)}>+</button>
          </>
        )}
      </ProductCard>
    );
    let tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
