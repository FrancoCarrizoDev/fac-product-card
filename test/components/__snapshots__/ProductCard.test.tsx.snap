// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`ProductTitle Debe de mostrar el componente correctament 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
</div>
`;
